lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -370,19 +334,11 @@ public function parse()
                 $treeWalkerChain->addTreeWalker($walker);
             }
 
-            switch (true) {
-                case $AST instanceof AST\UpdateStatement:
-                    $treeWalkerChain->walkUpdateStatement($AST);
-                    break;
-
-                case $AST instanceof AST\DeleteStatement:
-                    $treeWalkerChain->walkDeleteStatement($AST);
-                    break;
-
-                case $AST instanceof AST\SelectStatement:
-                default:
-                    $treeWalkerChain->walkSelectStatement($AST);
-            }
+            match (true) {
+                $AST instanceof AST\UpdateStatement => $treeWalkerChain->walkUpdateStatement($AST),
+                $AST instanceof AST\DeleteStatement => $treeWalkerChain->walkDeleteStatement($AST),
+                default => $treeWalkerChain->walkSelectStatement($AST),
```

Comment:
Let's be explicit about the remaining case that we expect here. Static analysis should recognize that the `match` statement covers all cases without a `default` and will notify us if that will ever change in the future. 💪🏻 

```suggestion
                $AST instanceof AST\SelectStatement => $treeWalkerChain->walkSelectStatement($AST),
```

Reply:
Looks nicer indeed 👍